src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -156,7 +156,7 @@ internal static BindingDataResolver GetAppSettingResolver(string originalValue,
                 // It's important that we only log the attribute property name, not the actual value to ensure
                 // that in cases where users accidentally use a secret key *value* rather than indirect setting name
                 // that value doesn't get written to logs.
-                throw new InvalidOperationException($"Unable to resolve value for property '{propInfo.DeclaringType.Name}.{propInfo.Name}'.");
+                throw new InvalidOperationException($"Unable to resolve app setting for property '{propInfo.DeclaringType.Name}.{propInfo.Name}'. Make sure the app setting exists and has a valid value.");
```

Comment:
It can - the error is written in the context of web apps where these are essentially all Application Settings. I'm happy to improve the message if you have a suggestion.